packages/@webex/plugin-cc/src/AgentProfile/AgentProfile.ts

```diff
@@ -0,0 +1,108 @@
+import {AgentProfileResponse} from './types';
+import AgentProfileService from '../AgentProfileService/AgentProfileService';
+import {
+  DesktopProfileResponse,
+  ListAuxCodesResponse,
+  ListTeamsResponse,
+  UserResponse,
+} from '../AgentProfileService/types';
+import {WebexSDK} from '../types';
+import {attributes, page, pageSize} from '../AgentProfileService/constants';
+
+export default class AgentProfile {
+  ciUserId: string;
+  agentProfile: AgentProfileResponse;
+  webex: WebexSDK;
+  wccAPIURL: string;
+
+  constructor(ciUserId: string, webex: WebexSDK, wccAPIURL: string) {
+    this.webex = webex;
+    this.ciUserId = ciUserId;
+    this.agentProfile = {
+      userDetails: {} as UserResponse['body'],
+      agentDesktopProfile: {} as DesktopProfileResponse['body'],
+      teamsList: [] as ListTeamsResponse['body'],
+      auxCodesList: [] as ListAuxCodesResponse['body']['data'],
```

Comment:
Is there a reason we are initializing this here? Why not populate this in a later function?

Reply:
I thought it will be good to initialize this properties inside the constructor itself before the method invocation. I think Initializing an object inside a constructor is generally better than initializing it inside a method because constructors are designed to ensure that objects are properly initialized when they are created:

Reply:
I read it somewhere, don't remember where I read it. But yeah we can improve the code by not initializing objects inside the constructor. Code Updated!